PIP_COMPATIBILITY.md

```diff
@@ -0,0 +1,317 @@
+dependencies.
+
+Pre-releases are [notoriously difficult](https://pubgrub-rs-guide.netlify.app/limitations/prerelease_versions)
+to model, and are a frequent source of bugs in packaging tools. Even `pip`, which is viewed as a
+reference implementation, has a number of open questions around pre-release handling ([#12469](https://github.com/pypa/pip/issues/12469),
+[#12470](https://github.com/pypa/pip/issues/12470), etc.). uv's pre-release handling is
+_intentionally_ limited and _intentionally_ requires user opt-in for pre-releases, to ensure
+correctness.
+
+In the future, uv _may_ support pre-release identifiers in transitive dependencies. However, it's
+likely contingent on evolution in the Python packaging specifications. The existing PEPs [do not
+cover "dependency resolution"](https://discuss.python.org/t/handling-of-pre-releases-when-backtracking/40505/17)
+and are instead focused on behavior for a _single_ version specifier. Further, the [original intent](https://discuss.python.org/t/handling-of-pre-releases-when-backtracking/40505/20)
+of the spec was _not_ to support cases in which pre-releases are _required_ for resolution despite
+a lack of pre-release specifiers. As such, there are unresolved questions around the correct and
+intended behavior for pre-releases in the packaging ecosystem more broadly.
+
+## Local version identifiers
+
+uv does not implement spec-compliant handling of local version identifiers (e.g., `1.0.0+local`).
+Though local version identifiers are rare in published packages (and, e.g., disallowed on PyPI),
+they're common in the PyTorch ecosystem. uv's incorrect handling of local version identifiers
+may lead to resolution failures in some cases.
+
+In the future, uv will implement spec-compliant handling of local version identifiers.
+For more, see [#1855](https://github.com/astral-sh/uv/issues/1855).
+
+## Packages that exist on multiple indexes
+
+In both uv and `pip`, users can specify multiple package indexes from which to search for
+the available versions of a given package. However, uv and `pip` differ in how they handle
+packages that exist on multiple indexes.
+
+For example, imagine that a company publishes an internal version of `requests` on a private index
+(`--extra-index-url`), but also allow installing packages from PyPI by default. In this case, the
+private `requests` would conflict with the public [`requests`](https://pypi.org/project/requests/)
+on PyPI.
+
+When uv searches for a package across multiple indexes, it will iterate over the indexes in order
+(preferring the `--extra-index-url` over the default index), and stop searching as soon as it
+finds a match. This means that if a package exists on multiple indexes, uv will limit its
+candidate versions to those present in the first index that contains the package.
+
+(`pip`, meanwhile, will combine the candidate versions from all indexes, and select the best
+version from the combined set.)
```

Comment:
I'd also add that pip makes no guarantees about the order (e.g. https://github.com/pypa/pip/issues/5045#issuecomment-369521345)